src/main/java/com/answerdigital/academy/answerking/service/OrderService.java

```diff
@@ -56,60 +56,64 @@ public Order updateOrder(final Long orderId, final OrderRequest orderRequest) {
     }
 
     @Transactional
-    public Order addItemToBasket(final Long orderId, final Long itemId, final Integer quantity) {
+    public Order addProductToBasket(final Long orderId, final Long productId, final Integer quantity) {
         final  Order order = findById(orderId);
-        final Item item = itemService.findById(itemId);
+        final Product product = productService.findById(productId);
 
-        if (!item.getAvailable()) {
-            throw new ItemUnavailableException(String.format("The item with ID %d is not available.", item.getId()));
+        if (!product.getAvailable()) {
+            throw new ProductUnavailableException(String.format("The product with ID %d is not available.", product.getId()));
         }
 
-        final Optional<OrderItem> existingOrderItem = order.getOrderItems()
+        final Optional<LineItem> existingOrderProduct = order.getLineItems()
                 .stream()
-                .filter(orderItem -> orderItem.getItem() == item)
+                .filter(lineItem -> lineItem.getProduct() == product)
                 .findFirst();
 
-        if (existingOrderItem.isPresent()) {
-            throw new ConflictException(String.format("Item id %s is already in the basket", item.getId()));
+        if (existingOrderProduct.isPresent()) {
+            throw new ConflictException(String.format("Product id %s is already in the basket", product.getId()));
         }
 
-        final OrderItem orderItem = new OrderItem(order, item, quantity);
-        order.getOrderItems().add(orderItem);
+        final LineItem lineItem = new LineItem(order, product, quantity);
+        order.getLineItems().add(lineItem);
 
         return orderRepository.save(order);
     }
 
-    public Order updateItemQuantity(final Long orderId, final Long itemId, final Integer itemQuantity) {
+    public Order updateProductQuantity(final Long orderId, final Long productId, final Integer productQuantity) {
         final Order order = findById(orderId);
-        final Item item = itemService.findById(itemId);
+        final Product product = productService.findById(productId);
 
-        final Optional<OrderItem> existingOrderItem = order.getOrderItems()
+        final Optional<LineItem> existingLineItem = order.getLineItems()
                 .stream()
-                .filter(orderItem -> orderItem.getItem() == item)
+                .filter(lineItem -> lineItem.getProduct() == product)
                 .findFirst();
 
-        if (existingOrderItem.isEmpty()) {
-            throw new NotFoundException(String.format("Item id = %s is not in the basket of order id = %s", orderId, itemId));
+        if (existingLineItem.isEmpty()) {
+            throw new NotFoundException(
+                    String.format("Product id = %d is not in the basket of order id = %d", orderId, productId)
+            );
         }
```

Comment:
Was changing '%d' to '%s' intentional? If so, what's the difference?

Reply:
Hi @MohammadIqbalAD - I changed it to %d from %s as %s is shorthand for any type (more specifically though used for Strings) and %d is used for integer types (incl. byte, short, int, long, bigint). As orderId and itemId are both numeric, I opted for %d.

Reply:
Ah makes sense now. Thanks @martintaylor1635